File: Social_Topic_Insight_3/models/task.py
from typing import List, Dict, Optional
from datetime import datetime
from pydantic import Field
from .base import MongoModel

class CrawlerTask(MongoModel):
    name: str
    platforms: List[str]            # ["weibo", "douyin"]
    keywords: List[str] = []        # 预测模式下的关键词
    mode: str = "hot_list"          # hot_list 或 prediction
    
    date_range: Dict[str, Optional[datetime]] = Field(default_factory=dict)
    
    status: str = "pending"         # pending, running, completed, failed
    progress: int = 0
    
    create_time: datetime = Field(default_factory=datetime.now)
    finished_time: Optional[datetime] = None
    
    log: str = ""                   # 实时日志
    result_summary: Dict = Field(default_factory=dict)

    class Config:
        collection_name = "crawler_tasks"

File: Social_Topic_Insight_spring/common/src/main/java/com/common/entity/Task.java
package com.common.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.util.Date;
import java.util.List;

/**
 * 任务实体（Mongo: crawler_tasks）。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "crawler_tasks") // 映射 Mongo 集合 crawler_tasks
public class Task {

    /** Mongo ObjectId（字符串形式）。 */
    @MongoId(targetType = FieldType.OBJECT_ID)
    private String id;

    /** 任务名称。 */
    private String name;

    /** 任务平台列表。 */
    private List<String> platforms;

    /** 任务模式。 */
    private String mode;

    /** 关键词列表（预测模式使用）。 */
    private List<String> keywords;

    /** 任务状态（pending/running/completed/failed）。 */
    private String status;

    /** 实时日志。 */
    private String log;

    /** 创建时间。 */
    @Field("create_time")
    private Date createTime;

    /** 完成时间。 */
    @Field("finished_time")
    private Date finishedTime;
}
